test/bin.js

```diff
@@ -28,11 +29,13 @@ describe('bin', function() {
       }
       consoleLog.apply(null, arguments)
     }
+    process.exit = function noop() {}
     process.argv = process.argv.slice(0, 2)
   })
 
   afterEach(function() {
     console.log = consoleLog // eslint-disable-line no-console
+    process.exit = processExit
     // purge yargs cache
     delete require.cache[require.resolve('yargs')]
```

Comment:
Are you suggesting something like below?

```diff
var assert = require('assert')
-var proxyquire = require('proxyquire')
+var proxyquire = require('proxyquire').noPreserveCache();
 var sinon = require('sinon')

 var consoleLog = console.log // eslint-disable-line no-console
@@ -19,6 +19,9 @@ var stub = {
       getUnusedRules: getUnusedRules,
     }
   },
+  'yargs': {
+    '@global': true
+  }
 }

 describe('bin', function() {
@@ -36,8 +39,6 @@ describe('bin', function() {
   afterEach(function() {
     console.log = consoleLog // eslint-disable-line no-console
     process.exit = processExit
-    // purge yargs cache
-    delete require.cache[require.resolve('yargs')]
   })

   it('option -c|--current', function() {
```

could not get it working, looks like `yargs` is cached after the first `require`. Let me know if I have mis interpreted.